### ISSUE TITLE
[HOLD] avoid duplicate errors in AuditResults

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/preservation_catalog#1353]
While working on #1352, some of the tests produced multiple errors for PreservedObject not found or for CompleteMoab not found.  We should de-dup the AuditResults.result_array.  To make it especially fun, it's an array of hashes.

**Comment by ndushay** [comment]
possibly blocked by PR #1352, but probably not.

**Comment by ndushay** [comment]
#1352 may be undergoing a huge rewrite, so this may not be necessary

**Comment by ndushay** [comment]
This is moot as, PR #1352 has been superseded by PR #1355
